src/onnxruntime.cc

```diff
@@ -1661,11 +1679,10 @@ ModelInstanceState::ReadOutputTensors(
   }
 
   size_t output_count = 0;
-  OrtValue** output_values = nullptr;
   RESPOND_ALL_AND_RETURN_IF_ORT_ERROR(
       responses, request_count,
       ort_api->GetBoundOutputValues(
-          io_binding_, default_allocator_, &output_values, &output_count));
+          io_binding_, default_allocator_, &output_buffer_, &output_count));
```

Comment:
Aren't you overwriting output_buffer_ here each time? Are you relying on it always returning the same buffer? Is that a valid assumption?

Reply:
I dont exactly understand your concern here
GetBoundOutputValues call will allocate the output buffer using the specified instance of the allocator which we then free before exiting ProcessRequest.

Reply:
I see, I missed that call to ReleaseOrtRunResources

Reply:
The only thing changing in this PR is now we are rightfully freeing the output_buffer_ before exiting ProcessRequests. Earlier we were not and hence the mem leak.

Reply:
OK, can you initiliaze output_buffer_ and I will start the CI